todo-mvc-kitchensink/src/todoProcesses.ts

```diff
@@ -0,0 +1,135 @@
+import { CommandRequest, createProcess } from '@dojo/stores/process';
+import { PatchOperation } from '@dojo/stores/state/Patch';
+import { add, remove, replace } from '@dojo/stores/state/operations';
+import uuid from '@dojo/core/uuid';
+
+export interface Todo {
+	id: string;
+	label: string;
+	completed?: boolean;
+	editing?: boolean;
+}
+
+export interface Todos {
+	[key: string]: Todo;
+}
+
+function addTodoCommand({ get }: CommandRequest): PatchOperation[] {
+	const id = uuid();
+	const count = get('/todoCount');
+	const todo = { label: get('/currentTodo').trim(), id };
+
+	return todo.label ? [
+		add(`/todos/${id}`, todo),
+		replace('/currentTodo', ''),
+		replace('/todoCount', count + 1)
```

Comment:
Do you think that adding a todo should do anything other than actually add the todo? There must be multiple actions that affect the counts of todos (total, active and completed)?

Reply:
So it could be it's own command that runs after each of the actions that need the counts to be recalculated.

Reply:
I don't necessarily see a problem with the add operation also updating the count since it's more efficient to increment it this way than to count them all every time. I think you could also argue that adding an item without updating the count means that this command would put the store in an inconsistent state. I can also see the logic for separating the two processes though.

I'd lean towards leaving it like this but I can change it if you'd like.

Reply:
In my head the commands represent discreet decoupled operations on the store's state and a process represent a sequence of commands required to perform all changes for a complete business or UI interaction. For example these command should actually be called, `AddTodoAndClearCurrentTodoAndUpdateTodoCount` which makes me think it's doing too much for a command. 

Re the store being in an inconsistent state, as commands are designed to perform these small targeted changes, they could put the store in an "inconsistent" state but the full process never should and I think that's how we should be thinking.

We can leave it like this for the first pass and then discuss updating the data structure and process composition in a later update; The reason why I'd prefer to have this structure in the example is because it should be absolutely the recommended approach using stores which in my thoughts, is creating small commands that act as the building blocks for larger processes.

Reply:
Ok, makes sense. I didn't change the data structure because it still is rather convenient, but I separated out counting todos and updating the `completed` flag into their own commands and added them to the appropriate processes.